CMakeLists.txt

```diff
@@ -15,35 +15,18 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 endif()
 
 file(GLOB_RECURSE SOURCES "${CMAKE_SOURCE_DIR}/Osiris/*.cpp")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/imgui/GL/gl3w.c")
 list(REMOVE_ITEM SOURCES "${CMAKE_SOURCE_DIR}/Osiris/imgui/imgui_impl_dx9.cpp") 
 list(REMOVE_ITEM SOURCES "${CMAKE_SOURCE_DIR}/Osiris/imgui/imgui_impl_win32.cpp") 
 list(REMOVE_ITEM SOURCES "${CMAKE_SOURCE_DIR}/Osiris/Hooks/MinHook.cpp") 
 
-# FreeType sources
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/autofit/autofit.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/cff/cff.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftbase.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftbitmap.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftdebug.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftglyph.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftinit.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftsynth.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/base/ftsystem.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/psaux/psaux.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/pshinter/pshinter.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/psnames/psnames.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/raster/raster.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/sfnt/sfnt.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/smooth/smooth.c")
-list(APPEND SOURCES "${CMAKE_SOURCE_DIR}/Osiris/freetype/src/truetype/truetype.c")
-
-add_compile_definitions(FT2_BUILD_LIBRARY)
-
-include_directories(/usr/local/include /usr/include/SDL2/ "${CMAKE_SOURCE_DIR}/Osiris/freetype/")
-link_directories(/usr/local/lib)
+set(OpenGL_GL_PREFERENCE "LEGACY")
+
+find_package(Freetype REQUIRED)
+find_package(SDL2 REQUIRED)
+find_package(OpenGL REQUIRED)
 
 add_library(Osiris SHARED ${SOURCES})
 
-target_link_libraries(Osiris SDL2 GL dl)
+target_include_directories(Osiris PRIVATE ${SDL2_INCLUDE_DIRS} ${FREETYPE_INCLUDE_DIRS})
+target_link_libraries(Osiris PRIVATE SDL2 OpenGL::GL ${CMAKE_DL_LIBS} Freetype::Freetype)
```

Comment:
What about just linking with `GL` here as it was before without using OpenGL package?

Reply:
I can ask why you recently replaced all stl::find with stl::ranges::find so Osiris not more compiling with LLVM's libc++(which not yet support STL ranges). Let me guess that it's because Osiris is a self-education project for you and you trying to use most modern programming practices in your code. With this feelings i've coded CMakeLists.txt using modern CMake syntax. But sure, it's your project and if you feel that it should link against "-lGL" directly i can make this, but, in my opinion, my way is more correct and future-proof.

Reply:
> I can ask why you recently replaced all stl::find with stl::ranges::find so Osiris not more compiling with LLVM's libc++(which not yet support STL ranges).
We have a GitHub Action that uses clang-10 and compiles successfully.

Using `find_package()` and setting a variable is overcomplicating the process in my opinion.

Reply:
I was talking about [LLVM libc++](https://libcxx.llvm.org/), AFAIK it used by default only on OSX (where must be much more problems to run Osiris) and it should be especially enabled on Linux. I just test tested it and it broken due stl::ranges still not supported. Not a big deal anyway.